### DISCUSSION TITLE
Shadow DOM Support

### DISCUSSION THREAD

**Discussion by matt-greenfield-15** [facebook/stylex#273] [Q&A] [answered]
My styles are working as expected when mounting my react app in the main DOM but the styles do not load when I mount it in a shadow root.
Is this something that StyleX needs to add support for? Or is it more of an issue for my CSS Loader? I'm using Vite if it matters.
Thanks for your help.

**Comment by nmn** [comment, marked as answer]
This is an issue with Shadow DOM. When using Shadow DOM, you need to load the CSS file within the Shadow DOM. If you're using web components, you'll have to load the CSS file in every component. There is no way for Shadow DOM components to share one global CSS file.

**Reply by nmn** [reply]
This is not a StyleX issue, but I'm leaving the discussion topic open in case someone else can help you. Please provide more details if you want further help.

**Reply by matt-greenfield-15** [reply]
Thanks @nmn - I guess my question really is "How can I control how and where the CSS is injected (for both dev and build modes) when using vite"
I appreciate that might not actually be a concern of this library itself as there is no official vite plugin.
If I can control that, then I can easily inject the CSS in to the shadow root.

For extra context, I'm actually fine with having a single CSS file as I only have a single Shadow DOM. I'm building a 'widget' that is placed on third party sites and am using Shadow DOM purely for the blocking of CSS inheritance.

**Reply by nmn** [reply]
The best solution with Shadow DOM is to add a `link` tag to import the CSS file within the component itself. This would work for production builds.

I don't see a good way to achieve what you need in `dev` mode. In the next version of StyleX, you will be able to define your own custom module to handle dynamic Style injection. Maybe you can find a solution by using that.

------

Generally, you'd lose most of the benefits of using StyleX by using Shadow DOM. Why not just use Vanilla CSS in an inline `<style>` tag for just the shadow DOM part? It's a separate world anyway?

The best reason for using StyleX within Shadow DOM would be to override the styles of the component in a reliable and predictable way. But if that was the case, you'd be OK with style inheritance too!

------

I've given you the best solutions I can think of, but I strongly suggest not using StyleX within Shadow DOM. It's not a good fit.

**Reply by matt-greenfield-15** [reply]
Thank you again, lots to think about!
